src/common/ant-registry.ts

```diff
@@ -29,9 +29,16 @@ import {
 import { createAoSigner } from '../utils/ao.js';
 import { AOProcess, InvalidContractConfigurationError } from './index.js';
 
-type ANTRegistryNoSigner = ProcessConfiguration;
-type ANTRegistryWithSigner = WithSigner<ProcessConfiguration>;
-type ANTRegistryConfig = ANTRegistryNoSigner | ANTRegistryWithSigner;
+type ANTRegistryConfig = Required<ProcessConfiguration> & {
+  strict?: boolean;
+  hyperbeamUrl?: string;
+};
+
+type ANTRegistryNoSigner = ANTRegistryConfig;
+type ANTRegistryWithSigner = WithSigner<ANTRegistryConfig>;
+type ANTRegistryConfigOptionalSigner =
```

Comment:
nit: call this ANTRegistryConfig and the existing ANTRegistryConfig as ANTRegistryProcessConfig or similar

Reply:
https://github.com/ar-io/ar-io-sdk/pull/508/commits/0970a50d8f9baefe6224324251f28dc359cb7856